crates/metadata-store/src/omnipaxos/store.rs

```diff
@@ -0,0 +1,1232 @@
+            }
+
+            self.last_applied_index = last_decided_index;
+        }
+
+        Ok(DecidedEntriesResult::Continue)
+    }
+
+    fn handle_stop_sign(
+        &mut self,
+        stop_sign: StopSign,
+    ) -> Result<DecidedEntriesResult, DecidedEntriesError> {
+        trace!("Handling stop sign");
+
+        // remove the stop sign from the entries that are applied to the kv_storage because it is
+        // virtual entry that is not part of the log
+        let last_decided_index = self
+            .omni_paxos
+            .as_ref()
+            .expect("to be present")
+            .get_decided_idx()
+            - 1;
+        self.last_applied_index = last_decided_index;
+
+        let metadata = Self::deserialize_omni_paxos_config_metadata(stop_sign.metadata.as_ref());
+
+        if Self::is_member(self.own_member_id, &metadata.members) {
+            debug!(configuration_id = %stop_sign.next_config.configuration_id, "Continue as part of new configuration: {:?}", stop_sign.next_config.nodes);
+
+            let mut rocksdb_storage = self.omni_paxos.take().expect("be present").into_inner();
+
+            let new_cluster_config = stop_sign.next_config.clone();
+            let omni_paxos_configuration = OmniPaxosConfiguration {
+                own_member_id: self.own_member_id,
+                cluster_config: stop_sign.next_config,
+                members: metadata.members,
+            };
+
+            Self::reset_storage_for_new_configuration(
+                &mut rocksdb_storage,
+                &omni_paxos_configuration,
+                last_decided_index,
+            )?;
+
+            let omni_paxos = Self::create_omni_paxos(
+                omni_paxos_configuration.own_member_id.node_id,
+                omni_paxos_configuration.cluster_config,
+                rocksdb_storage,
+            );
+
+            self.omni_paxos = Some(omni_paxos);
+            self.cluster_config = new_cluster_config;
+            self.members = omni_paxos_configuration.members;
+
+            self.answer_join_callbacks();
+            self.check_leadership();
+
+            self.update_node_addresses(&Metadata::with_current(|m| m.nodes_config_ref()));
+
+            Ok(DecidedEntriesResult::Continue)
+        } else {
+            debug!(configuration_id = %stop_sign.next_config.configuration_id, "Stopping since I am no longer part of the new configuration.");
+
+            let mut rocksdb_storage = self.omni_paxos.take().expect("to be present").into_inner();
+
+            // remember the latest configuration we have seen for future checks
+            Self::reset_storage_for_new_configuration(
+                &mut rocksdb_storage,
+                &OmniPaxosConfiguration {
+                    own_member_id: self.own_member_id,
+                    cluster_config: stop_sign.next_config,
+                    members: metadata.members,
+                },
+                last_decided_index,
+            )?;
```

Comment:
This might be helpful if this node is removed from the next configuration but some other nodes ask this node whether it knows which nodes are part of the new configuration and what the log is.